Traceback (most recent call last):
  File "c:\python39\lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "c:\python39\lib\site-packages\nbclient\client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "C:\Users\matth\AppData\Roaming\Python\Python39\site-packages\jupyter_core\utils\__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
  File "c:\python39\lib\asyncio\base_events.py", line 642, in run_until_complete
    return future.result()
  File "c:\python39\lib\site-packages\nbclient\client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "c:\python39\lib\site-packages\nbclient\client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "c:\python39\lib\site-packages\nbclient\client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Converting RGB image to grayscale
gray = cv2.cvtColor(image1, cv2.COLOR_BGR2GRAY)

# Calculate the contours from binary image
canny = cv2.Canny(gray, 50, 200)

# Find contours
contours, hierarchy = cv2.findContours(canny, cv2.RETR_TREE, cv2.CHAIN_APPROX_NONE)

# Draw contours
with_contours = cv2.drawContours(image1.copy(), contours, -1, (0, 255, 0), 2)

# Show
plt.subplot(1, 4, 1)
plt.imshow(image1)
plt.subplot(1, 4, 2)
plt.imshow(gray, cmap='gray')
plt.subplot(1, 4, 3)
plt.imshow(canny, cmap='gray')
plt.subplot(1, 4, 4)
plt.imshow(with_contours)
plt.show()

------------------


[1;31m---------------------------------------------------------------------------[0m
[1;31merror[0m                                     Traceback (most recent call last)
Cell [1;32mIn[3], line 2[0m
[0;32m      1[0m [38;5;66;03m# Converting RGB image to grayscale[39;00m
[1;32m----> 2[0m gray [38;5;241m=[39m [43mcv2[49m[38;5;241;43m.[39;49m[43mcvtColor[49m[43m([49m[43mimage1[49m[43m,[49m[43m [49m[43mcv2[49m[38;5;241;43m.[39;49m[43mCOLOR_BGR2GRAY[49m[43m)[49m
[0;32m      4[0m [38;5;66;03m# Calculate the contours from binary image[39;00m
[0;32m      5[0m canny [38;5;241m=[39m cv2[38;5;241m.[39mCanny(gray, [38;5;241m50[39m, [38;5;241m200[39m)

[1;31merror[0m: OpenCV(4.8.0) D:\a\opencv-python\opencv-python\opencv\modules\imgproc\src\color.cpp:182: error: (-215:Assertion failed) !_src.empty() in function 'cv::cvtColor'


